### PR TITLE
Move to the quay.io

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@vparfonov: FYI: [Happy path E2E tests](https://github.com/eclipse/che/pull/15569#issuecomment-568708510) failed because Eclipse Che had not properly deployed to minikube:
https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Multiuser-Che-PR-check-E2E-Happy-path-tests-against-k8s/1501/artifact/logs-and-configs/che-logs/che-778888c64d-xvc4r.pod.log/*view*/
```
24-Dec-2019 09:40:18.588 INFO [jgroups-6,RemoteSubscriptionChannel,che-778888c64d-xvc4r-39719] org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading Illegal access: this web application instance has been stopped already. Could not load [org.jgroups.util.Headers]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
	java.lang.IllegalStateException: Illegal access: this web application instance has been stopped already. Could not load [org.jgroups.util.Headers]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
		at org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading(WebappClassLoaderBase.java:1380)
		at org.apache.catalina.loader.WebappClassLoaderBase.checkStateForClassLoading(WebappClassLoaderBase.java:1368)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1220)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1182)
		at org.jgroups.Message.putHeader(Message.java:466)
		at org.jgroups.protocols.pbcast.STABLE.sendStableMessage(STABLE.java:667)
		at org.jgroups.protocols.pbcast.STABLE$StableTask.run(STABLE.java:832)
		at org.jgroups.util.TimeScheduler3$Task.run(TimeScheduler3.java:328)
		at org.jgroups.util.TimeScheduler3$RecurringTask.run(TimeScheduler3.java:362)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
24-Dec-2019 09:40:18.589 INFO [jgroups-6,RemoteSubscriptionChannel,che-778888c64d-xvc4r-39719] org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading Illegal access: this web application instance has been stopped already. Could not load [ch.qos.logback.core.status.WarnStatus]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
	java.lang.IllegalStateException: Illegal access: this web application instance has been stopped already. Could not load [ch.qos.logback.core.status.WarnStatus]. The following stack trace is thrown for debugging purposes as well as to attempt to terminate the thread which caused the illegal access.
		at org.apache.catalina.loader.WebappClassLoaderBase.checkStateForResourceLoading(WebappClassLoaderBase.java:1380)
		at org.apache.catalina.loader.WebappClassLoaderBase.checkStateForClassLoading(WebappClassLoaderBase.java:1368)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1220)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1182)
		at ch.qos.logback.classic.LoggerContext.noAppenderDefinedWarning(LoggerContext.java:186)
		at ch.qos.logback.classic.Logger.callAppenders(Logger.java:264)
		at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
		at ch.qos.logback.classic.Logger.filterAndLog_0_Or3Plus(Logger.java:383)
		at ch.qos.logback.classic.Logger.error(Logger.java:538)
		at org.jgroups.logging.Slf4jLogImpl.error(Slf4jLogImpl.java:104)
		at org.jgroups.util.TimeScheduler3$Task.run(TimeScheduler3.java:331)
		at org.jgroups.util.TimeScheduler3$RecurringTask.run(TimeScheduler3.java:362)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
```

**Review by sleshchenko — APPROVED** [review]
:tada: 

Is there any plan to update values in chectl as well? For example: https://github.com/che-incubator/chectl/blob/5eb787953dd4eb67259ee141d147b99632799315/src/constants.ts
https://github.com/che-incubator/chectl/blob/72a098dbee61ec571d93306317c19da4de912770/make-release.sh

**Comment by vparfonov** [comment]
> 🎉
> 
> Is there any plan to update values in chectl as well? For example: https://github.com/che-incubator/chectl/blob/5eb787953dd4eb67259ee141d147b99632799315/src/constants.ts
> https://github.com/che-incubator/chectl/blob/72a098dbee61ec571d93306317c19da4de912770/make-release.sh
 I will create PR

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties [hunk 1]
```diff
@@ -545,12 +545,12 @@ che.factory.default_plugins=eclipse/che-machine-exec-plugin/nightly
 # Default Editor that should be provisioned into Devfile if there is no specified Editor
 # Format is `editorPublisher/editorName/editorVersion` value.
 # `NULL` or absence of value means that default editor should not be provisioned.
-che.workspace.devfile.default_editor=eclipse/che-theia/next
+che.workspace.devfile.default_editor=quay.io/eclipse/che-theia/next
```
Comment (skabashnyuk) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties]: Not sure about this change. This is id of an editor, not the image id.

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties [hunk 2]
```diff
@@ -545,12 +545,12 @@ che.factory.default_plugins=eclipse/che-machine-exec-plugin/nightly
...
 
 # Default Plugins which should be provisioned for Default Editor.
 # All the plugins from this list that are not explicitly mentioned in the user-defined devfile
 # will be provisioned but only when the default editor is used or if the user-defined editor is
 # the same as the default one (even if in different version).
 # Format is comma-separated `pluginPublisher/pluginName/pluginVersion` values, for example
 # eclipse/che-theia-exec-plugin/0.0.1,eclipse/che-theia-terminal-plugin/0.0.1
-che.workspace.devfile.default_editor.plugins=eclipse/che-machine-exec-plugin/nightly
+che.workspace.devfile.default_editor.plugins=quay.io/eclipse/che-machine-exec-plugin/nightly
```
Comment (skabashnyuk) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties]: same as editor

**Review by skabashnyuk — DISMISSED** [review]

**Comment by skabashnyuk** [comment]
Do you plan to update images in https://github.com/eclipse/che-plugin-registry/tree/master/v3/plugins?

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties (hunk 1 above)
Comment (sleshchenko) [reply assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties]: you're totally right. 
Images are defined in Plugin Registry and it's already changed to quay.io in https://github.com/eclipse/che-plugin-registry/pull/323

**Pull request merged**